### DISCUSSION TITLE
Lista de ideas de videojuegos

### DISCUSSION THREAD

**Discussion by G00Z-G00Z** [G00Z-G00Z/Timpulsa-Curso-JS#14] [Ideas]
Aqui se pone una lista de posibles videojuegos para desarrollar

**Comment by G00Z-G00Z** [comment]
El juego de la serpiente. 
![image](https://user-images.githubusercontent.com/73507446/133913196-c5a56f5c-690d-496c-84f1-9086e54bff42.png)
Creo que sería muy fácil programar la serpiente, el problema es detectar cuando pierde y como moverla (ya que los puntos pasados tienen que serguir el punto anterior) 
Podríamos hacer una lista de posición pasadas y que un cuadro sea la serpiente

**Comment by G00Z-G00Z** [comment]
# Juego multijugador

![image](https://user-images.githubusercontent.com/73507446/133913225-b60adf60-1cd9-489f-93be-f690ce74a5c2.png)

Es un juego en el que meteoritos aparecen a los lados, y ambos jugadores compiten por llegar hasta arriba. Creo que estaría padre, pero como las clases son el linea a lo mejor no jalaría. Pero creo que un juego multijugador sería bueno. 

[Link del gif](https://www.google.com/url?sa=i&url=https%3A%2F%2Fdev.to%2Flukegarrigan%2Ftop-5-best-games-to-code-as-a-beginner-9n&psig=AOvVaw0lzRi3RUfdJkJOoFy9sLIr&ust=1632103868311000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCJjv4oH7ifMCFQAAAAAdAAAAABAP)

**Reply by G00Z-G00Z** [reply]
[Link para ideas de videojuegos](https://dev.to/lukegarrigan/top-5-best-games-to-code-as-a-beginner-9n)

Incluye el **space race** que discutí antes: 
> # Juego multijugador
> ![image](https://user-images.githubusercontent.com/73507446/133913225-b60adf60-1cd9-489f-93be-f690ce74a5c2.png)
> 
> Es un juego en el que meteoritos aparecen a los lados, y ambos jugadores compiten por llegar hasta arriba. Creo que estaría padre, pero como las clases son el linea a lo mejor no jalaría. Pero creo que un juego multijugador sería bueno.
> 
> [Link del gif](https://www.google.com/url?sa=i&url=https%3A%2F%2Fdev.to%2Flukegarrigan%2Ftop-5-best-games-to-code-as-a-beginner-9n&psig=AOvVaw0lzRi3RUfdJkJOoFy9sLIr&ust=1632103868311000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCJjv4oH7ifMCFQAAAAAdAAAAABAP)

**Reply by G00Z-G00Z** [reply]
A lo mejor no estaría chido porque como están solos en sus casas, puede que no tengan con quien jugar

**Comment by G00Z-G00Z** [comment]
Tetris

Obviamente

![image](https://user-images.githubusercontent.com/73507446/133913258-0bb941f8-29b2-4a32-b2f9-69993ab54b9e.png)

**Comment by G00Z-G00Z** [comment]
Simón dice

![image](https://user-images.githubusercontent.com/73507446/133913304-9568fcb1-e238-4672-b5d1-5053fdfa7fe5.png)

**Comment by G00Z-G00Z** [comment]
Podemos hacer un platformer tipo kirby, con unos 3 niveles.
![image](https://user-images.githubusercontent.com/73507446/133913342-134ded91-d8e7-41c5-a477-03ebcb705607.png)

Que sea una mecánica de ataque y de enemigos sencillos

**Comment by G00Z-G00Z** [comment]
# The imposible game
![image](https://user-images.githubusercontent.com/73507446/133913376-042c8a3d-8bf5-4824-84b9-9b0eb4a3a5f2.png)

**Comment by G00Z-G00Z** [comment]
[![Imagen de crossyrode](https://play-lh.googleusercontent.com/V31nfvzNcu_47lr8y2L7m3uw9azrpGyox88gpzhQxwfZSXliFJmIRVSlYJL1eOo_X2U)](https://www.youtube.com/watch?v=a3pTw0jmxlg)
_Dale click para ir al video_

# Crossy Road en 2D

Podríamos hacer un crossy road en 2d. Siento que el código se parecería a:

> # Juego multijugador
> ![image](https://user-images.githubusercontent.com/73507446/133913225-b60adf60-1cd9-489f-93be-f690ce74a5c2.png)
> 
> Es un juego en el que meteoritos aparecen a los lados, y ambos jugadores compiten por llegar hasta arriba. Creo que estaría padre, pero como las clases son el linea a lo mejor no jalaría. Pero creo que un juego multijugador sería bueno.
> 
> [Link del gif](https://www.google.com/url?sa=i&url=https%3A%2F%2Fdev.to%2Flukegarrigan%2Ftop-5-best-games-to-code-as-a-beginner-9n&psig=AOvVaw0lzRi3RUfdJkJOoFy9sLIr&ust=1632103868311000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCJjv4oH7ifMCFQAAAAAdAAAAABAP)

**Reply by G00Z-G00Z** [reply]
![image](https://user-images.githubusercontent.com/73507446/135166841-0d38fe41-e495-4fbc-af82-53232a409553.png)
Color switch

**Reply by G00Z-G00Z** [reply]
Ir más por el Color switch

**Comment by G00Z-G00Z** [comment]
[![portal 2 juego](https://static.wikia.nocookie.net/halflife7283/images/f/f9/Portal2cover.jpg/revision/latest/top-crop/width/360/height/450?cb=20150618192608&path-prefix=es)](https://www.crazygames.com/game/portal-2d)
_Hazme click para ir a ver el juego_

# Portal 2d

Encontré una versión de portal en 2d !! Dale una checada. Creo que podríamos imitar el juego bastante bien!!

**Comment by G00Z-G00Z** [comment]
![Angry birds](https://static.wikia.nocookie.net/videojuego/images/2/2d/Angry_Birds_promo_art.png/revision/latest?cb=20141022144544)

# Angry birds

Tendríamos que ver bien como funciona el engine de física...

**Comment by G00Z-G00Z** [comment]
[![Mini golf](https://img.youtube.com/vi/RJJugb2uJPs/0.jpg)](https://www.youtube.com/watch?v=RJJugb2uJPs)
_clickeame para ir al video_
# Minigolf

Es un tipo de mini golf, en el que tienes que llevar una pelota de un lugar a otro. Es muy sencillo y Tech with tim, tiene un tutorial. Lo único que podría ser un problema, es la física (pero como `kaboomjs` ya implementa la física, solamente tendríamos que concentrarnos en el tipo de suelo y así

**Comment by G00Z-G00Z** [comment]
![Piano tiles](https://is5-ssl.mzstatic.com/image/thumb/Purple125/v4/8c/ce/7e/8cce7e12-39cf-b3e0-1eb8-eb473e12f7b0/AppIcon-1x_U007emarketing-0-7-0-0-85-220.png/230x0w.webp)

# Piano tiles

Sería muy sencillo de programar, pero personalmente, a mi no me gustan. A parte, como el juego tiene que ver con música, a lo mejor es un poco más difícil

**Comment by G00Z-G00Z** [comment]
# Choose your own adventure

Una aventura en base a texto, y con esto podemos hacer un arbol de desiciones muy grande. Puede ser que nomás sea enseñar texto y así. Lo único, es que casi no tiene interactividad. Pero creo que los chavos pueden explotar su creatividad (creo que lo podríamos sacar en una sola clase para enseñar "escenas")

**Reply by G00Z-G00Z** [reply]
Para enseñar escenas

**Comment by G00Z-G00Z** [comment]
# 2 truths and 1 lie

Juego donde los jugadores ponen 2 truths and 1 lie, y el otro jugador tiene que adivinar cual es la mentira. Con esto podríamos hacer un score system, y podríamos hacer que sea un poco más interactivo (Sin embargo, como solamente están en casa, a lo mejor no queda tan bien)

**Comment by GilVa22** [comment]
![cookie clicker](https://play-lh.googleusercontent.com/OssE3ON9WsLZedOF39UCgtIHcRYfV0OqQS9O78LfmRdxSyKdHX52G2OFa0LkG6D-k9w)

# Cookie Clicker

Mas que nada por el tipo de juego, hacer un clicker debe estar sencillo y es puro manejo de puntos, podemos usarlo de ejemplo sencillo, con dos o tres mejoras nadamás.

**Reply by G00Z-G00Z** [reply]
Bueno para enseñar sistemas de puntajes

**Comment by GilVa22** [comment]
![megaman](https://i.ytimg.com/vi/C9t5uPrRyig/hqdefault.jpg)

# MegaMan

Retomando la idea que mencionabas del kirby, otro juego de plataformas con enemigos que podríamos hacer es megaman.

**Comment by GilVa22** [comment]
![DoodleJump](https://lh3.googleusercontent.com/8NdduY243jizXkxVv5D1mKTON_-NTaraBCkNwMYWHaZskwnUQfxfERtUqQGmvSr2ionxAVmOgn3jAEcl7W6_2Hfd-w=w640-h400-e365-rj-sc0x00ffffff)

# Doodle jump

La verdad no se que tan complicado pueda ser, porque si tiene algunos detallitos como los tipos de plataforma y que se generan infinitamente, pero creo que se puede hacer un nivel sencillo con un principio y un final.

**Comment by G00Z-G00Z** [comment]
# Temas de videojuegos

1. Interacciones con el teclado
2. Interacciones con el mouse
3. El sistema de body() (gravedad, colisiones, platformer)
4. Escenas
5. Manejo de metadatos del videojuego (puntaje, mensajes, etc) 
6. Manjear ui (layers, destruccion de objetos, etc)

**Comment by G00Z-G00Z** [comment]
# Último videojuego

Tiene que ser un **platformer** para aprovechar el engine de `kaboomjs`
